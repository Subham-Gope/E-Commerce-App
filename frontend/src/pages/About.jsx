import { assets } from "../assets/assets";
import NewsLetterBox from "../components/NewsLetterBox";
import Title from "../components/Title";

const About = () => {
  return (
    <div>
      <div className="text-2xl text-center pt-8 border-t">
        <Title text1={"ABOUT"} text2={"US"} />
      </div>
      <div className="my-10 flex flex-col md:flex-row gap-16">
        <img
          className="w-full md:max-w-[450px]"
          src={assets.about_img}
          alt="about-image"
        />
        <div className="flex flex-col justify-center gap-6 md:w-2/4 text-gray-6">
          <p>
            Forever brings you timeless fashion and lifestyle essentials crafted
            to elevate your everyday. Explore premium quality, sustainable
            designs that seamlessly blend style, comfort, and durability for all
            seasons.
          </p>
          <p>
            Rediscover shopping with Forever! Our curated collections cater to
            every taste, ensuring you stay trendy while embracing
            sustainability. Experience effortless elegance with products
            designed to last a lifetime.
          </p>
          <b className="text-gray-800">Our Mission</b>
          <p>
            Our mission at Forever is to create timeless, sustainable designs
            that inspire confidence and celebrate individuality. We aim to
            deliver unmatched quality while championing ethical practices and
            environmental responsibility.
          </p>
        </div>
      </div>

      <div className="text-xl py-4">
        <Title text1={"WHY"} text2={"CHOOSE US"} />
      </div>

      <div className="flex flex-col md:flex-row text-sm mb-20">
        <div className="border px-10 md:px-16 py-8 sm:py-20 flex flex-col gap-5">
          <b>Quality Assurance: </b>
          <p className="text-gray-600">
            Our mission in Quality Assurance is to ensure every product at
            Forever meets the highest standards of excellence. We are committed
            to delivering reliability, durability, and customer satisfaction in
            every detail.
          </p>
        </div>
        <div className="border px-10 md:px-16 py-8 sm:py-20 flex flex-col gap-5">
          <b>Convenience: </b>
          <p className="text-gray-600">
            At Forever, we prioritize convenience by offering a seamless
            shopping experience. From intuitive navigation to fast delivery, we
            make it easy to find and enjoy the products you love.
          </p>
        </div>
        <div className="border px-10 md:px-16 py-8 sm:py-20 flex flex-col gap-5">
          <b>Exceptional Customer Service: </b>
          <p className="text-gray-600">
            Forever is dedicated to providing exceptional customer service. Our
            team is here to support you every step of the way, ensuring your
            experience is seamless, personalized, and truly satisfying.
          </p>
        </div>
      </div>
      <NewsLetterBox />
    </div>
  );
};

export default About;
