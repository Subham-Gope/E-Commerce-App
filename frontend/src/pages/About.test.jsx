import { describe, expect, test, vi } from "vitest";
import { screen, render } from "@testing-library/react";
import * as assetsModule from "../assets/assets";
import About from "./About";

// Mocking the title component
vi.mock("../components/Title", () => ({
  default: ({ text1, text2 }) => (
    <h2 data-testid="title">
      {text1} {text2}
    </h2>
  ),
}));

// Mocking the Newaletter component
vi.mock("../components/NewsLetterBox", () => ({
  default: () => {
    return <div data-testid="newsletter-box">Newsletter Box</div>;
  },
}));

describe("Testing the About.jsx file", () => {
  test("renders the about page along with the text and newletter component", () => {
    // mocking the assets object to control the image used in the component
    vi.spyOn(assetsModule, "assets", "get").mockReturnValue({
      about_img: "mock-about-image.png",
    });

    // rendering and setting up the snapshot matching
    const { container } = render(<About />);
    expect(container).toMatchSnapshot();

    // checking the title component mock is rendered
    const titles = screen.getAllByTestId(/title/i);
    expect(titles).toBeTruthy();
    expect(titles).toHaveLength(2);
    expect(titles[0]).toHaveTextContent(/about us/i);
    expect(titles[1]).toHaveTextContent(/why choose us/i);

    // checking the images are rendered
    const aboutImage = screen.getByAltText(/about-image/i);
    expect(aboutImage).toBeInTheDocument();
    expect(aboutImage).toHaveAttribute("src", "mock-about-image.png");

    // checking the  descriptions are displayed
    expect(
      screen.getByText(/forever brings you timeless fashion and lifestyle/i)
    ).toBeInTheDocument();
    expect(
      screen.getByText(/Rediscover shopping with Forever!/i)
    ).toBeInTheDocument();
    expect(
      screen.getByText(/Our mission at Forever is to create timeless/i)
    ).toBeInTheDocument();

    // checking the titles and descriptions
    expect(screen.getByText(/Quality Assurance:/i)).toBeInTheDocument();
    expect(
      screen.getByText(/Our mission in Quality Assurance is to ensure/i)
    ).toBeInTheDocument();

    expect(screen.getByText(/Convenience:/i)).toBeInTheDocument();
    expect(
      screen.getByText(
        /At Forever, we prioritize convenience by offering a seamless/i
      )
    ).toBeInTheDocument();

    expect(
      screen.getByText(/Exceptional Customer Service:/i)
    ).toBeInTheDocument();
    expect(
      screen.getByText(/Forever is dedicated to providing exceptional/i)
    ).toBeInTheDocument();

    // checking the newsletter component is rendered

    const newsLetter = screen.getByTestId("newsletter-box");
    expect(newsLetter).toBeInTheDocument();
  });
});
