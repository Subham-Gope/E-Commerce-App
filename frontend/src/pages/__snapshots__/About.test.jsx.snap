// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the About.jsx file > renders the about page along with the text and newletter component 1`] = `
<div>
  <div>
    <div
      class="text-2xl text-center pt-8 border-t"
    >
      <h2
        data-testid="title"
      >
        ABOUT
         
        US
      </h2>
    </div>
    <div
      class="my-10 flex flex-col md:flex-row gap-16"
    >
      <img
        alt="about-image"
        class="w-full md:max-w-[450px]"
        src="mock-about-image.png"
      />
      <div
        class="flex flex-col justify-center gap-6 md:w-2/4 text-gray-6"
      >
        <p>
          Forever brings you timeless fashion and lifestyle essentials crafted to elevate your everyday. Explore premium quality, sustainable designs that seamlessly blend style, comfort, and durability for all seasons.
        </p>
        <p>
          Rediscover shopping with Forever! Our curated collections cater to every taste, ensuring you stay trendy while embracing sustainability. Experience effortless elegance with products designed to last a lifetime.
        </p>
        <b
          class="text-gray-800"
        >
          Our Mission
        </b>
        <p>
          Our mission at Forever is to create timeless, sustainable designs that inspire confidence and celebrate individuality. We aim to deliver unmatched quality while championing ethical practices and environmental responsibility.
        </p>
      </div>
    </div>
    <div
      class="text-xl py-4"
    >
      <h2
        data-testid="title"
      >
        WHY
         
        CHOOSE US
      </h2>
    </div>
    <div
      class="flex flex-col md:flex-row text-sm mb-20"
    >
      <div
        class="border px-10 md:px-16 py-8 sm:py-20 flex flex-col gap-5"
      >
        <b>
          Quality Assurance: 
        </b>
        <p
          class="text-gray-600"
        >
          Our mission in Quality Assurance is to ensure every product at Forever meets the highest standards of excellence. We are committed to delivering reliability, durability, and customer satisfaction in every detail.
        </p>
      </div>
      <div
        class="border px-10 md:px-16 py-8 sm:py-20 flex flex-col gap-5"
      >
        <b>
          Convenience: 
        </b>
        <p
          class="text-gray-600"
        >
          At Forever, we prioritize convenience by offering a seamless shopping experience. From intuitive navigation to fast delivery, we make it easy to find and enjoy the products you love.
        </p>
      </div>
      <div
        class="border px-10 md:px-16 py-8 sm:py-20 flex flex-col gap-5"
      >
        <b>
          Exceptional Customer Service: 
        </b>
        <p
          class="text-gray-600"
        >
          Forever is dedicated to providing exceptional customer service. Our team is here to support you every step of the way, ensuring your experience is seamless, personalized, and truly satisfying.
        </p>
      </div>
    </div>
    <div
      data-testid="newsletter-box"
    >
      Newsletter Box
    </div>
  </div>
</div>
`;
